x-pack/plugins/alerting/index.ts

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Root } from 'joi';
+import { init } from './server';
+import mappings from './mappings.json';
+
+export { AlertingPlugin, AlertsClient, AlertType, AlertExecuteOptions } from './server';
+
+export function alerting(kibana: any) {
+  return new kibana.Plugin({
+    id: 'alerting',
+    configPrefix: 'xpack.alerting',
+    require: ['kibana', 'elasticsearch', 'actions', 'task_manager'],
```

Comment:
Plugin doesn't rely on `encrypted_saved_objects` but I added recursively the plugin dependencies within `isEnabled` so that the plugin is disabled if ever `encrypted_saved_objects` is disabled.

Reply:
Yeah, I got it, it just looks a bit confusing that it doesn't "require" plugin, but if this dependency is disabled, this plugin is disabled too :)